### DISCUSSION TITLE
MartyPC, old screen simulation and CPU mods

### DISCUSSION THREAD

**Discussion by nico7550** [86Box/86Box#3452] [Ideas]
Hi,

You should have hear about MartyPC:
MartyPC IBM PC/XT emulator
https://int10h.org/blog/2023/07/martypc-pc-xt-emulator-raising-the-bar/
https://github.com/dbalsom/martypc
It could be nice to have some exchange with him ?

From there I found this:
ROM BIOS for PC, XT, 8088 Clone or Turbo PCs (including a one for 86box)
https://glabios.org/
https://github.com/640-KB/GLaBIOS

And this (from the MartyPC creator) (could be a nice feature mixed with screen bezels in fullscreen):
Oldschool Monitor Simulation with FFmpeg
https://www.youtube.com/watch?v=sZdc655YefA&ab_channel=x86VileR
https://www.youtube.com/watch?v=oMibMBPSO-o&ab_channel=x86VileR

In parallel, vswitchzero show socket 5/7 mods that could be nice to have in CPU settings:
https://www.youtube.com/@vswitchzero/videos

And CPU Galaxy show some mods too:
https://www.youtube.com/@CPUGalaxy/videos

That just some ideas and trails (and I'm sure you have a lot of them), nothing more, love 86Box, thanks

**Comment by dbalsom** [comment]
those monitor simulations are by VileR, not me.

**Reply by nico7550** [reply]
Sorry for the mistake and thanks for corrected me.
I just write to him, maybe somethig can be done...

Did the 86Box creator get in touch with you ?

**Reply by dbalsom** [reply]
i've been helping out where i can.

**Comment by viler-int10h** [comment]
> Sorry for the mistake and thanks for corrected me. I just write to him, maybe somethig can be done...

@nico7550: my CRT simulation is basically a script for ffmpeg, and entails lots of video transcoding using CPU-driven filters. It's very far from real-time, so it's really not suitable for emulation - fairly high-end rigs are lucky to manage 2%(!) realtime speed with it.

If I knew much GPU programming I'd try to port it to a shader, but existing ones already do a good enough job and are available in emulators like DOSBox(-X) and MAME, so those would be a much better starting point.  (I thought 86box had support as well - or maybe it was just PCem?)

**Reply by OBattler** [reply]
86Box does support shaders, though support is currently less complete than in PCem (ie. you can only use one shader at a time).

**Comment by nico7550** [comment]
Just see this for retroarch!
https://github.com/HyperspaceMadness/Mega_Bezel

I also like HDD and Floppy sound request ;-)
